### PR TITLE
Implement blade chassis

### LINKED ISSUE #34: Implement Blade Chassis
As oneview-redfish-toolkit, I access "/redfish/v1/Chassis/<server_hardware_uuid>" and receive basic information of the server hardware chassis and a link back to the ComputerSystem it is related to.

### PR DESCRIPTION
**Pull request by danielreed**
Closes #34

---
This change is [Reviewable](https://reviewable.io/reviews/hewlettpackard/oneview-redfish-toolkit/47)

### CODE REVIEW

File: oneview_redfish_toolkit/api/chassis.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+import collections
+from oneview_redfish_toolkit.api.redfish_json_validator \
+    import RedfishJsonValidator
+
+
+class Chassis(RedfishJsonValidator):
```
Comment (llvieira) [inline oneview_redfish_toolkit/api/chassis.py]: This class could be a super class populating only Redfish common values between Blade, Rack and Enclosure. Therefore will you create the Blade sub class into chassi.py module inheriting from Chassi class and its constructor populate with only its specific values.

Doing this, the call of _validate() should happen on sub classes constructor.

**Comment by llvieira** [comment]
Reviewed 2 of 7 files at r1, 5 of 5 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/47)*

**Comment by danielreed** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[oneview_redfish_toolkit/api/chassis.py, line 22 at r2](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/47#-KtIzp0p10cXn930ahjM-r2-22:-KtS4dAyDZy4l2XTQxu7:b-cfa3zy) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/be850b4afbba78abe3abc4340085037bac0261ea/oneview_redfish_toolkit/api/chassis.py#L22)):*
<details><summary><i>Previously, llvieira (Lucas Lima Vieira) wrote…</i></summary><blockquote>

This class could be a super class populating only Redfish common values between Blade, Rack and Enclosure. Therefore will you create the Blade sub class into chassi.py module inheriting from Chassi class and its constructor populate with only its specific values.

Doing this, the call of _validate() should happen on sub classes constructor.
</blockquote></details>

Do you want me to make this change in this patch?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/47)*

**Comment by llvieira** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[oneview_redfish_toolkit/api/chassis.py, line 22 at r2](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/47#-KtIzp0p10cXn930ahjM-r2-22:-KtS6pXUnjDuku1p6juI:b-s6edgs) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/be850b4afbba78abe3abc4340085037bac0261ea/oneview_redfish_toolkit/api/chassis.py#L22)):*
<details><summary><i>Previously, danielreed (Daniel Reed) wrote…</i></summary><blockquote>

Do you want me to make this change in this patch?
</blockquote></details>

I can make it when I implement Enclosure Chassi, if you want...

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/47)*

**Comment by llvieira** [comment]
:lgtm:

---

Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/47#-:-KtS7rD52lNB8Whmmxz8:bnfp4nl)*

**Comment by ricardoas** [comment]
Added a few comments... but overall it looks good!

---

Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[oneview_redfish_toolkit/api/chassis.py, line 22 at r2](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/47#-KtIzp0p10cXn930ahjM-r2-22:-KtmGUSI95-EkofHCMAn:bksprkg) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/be850b4afbba78abe3abc4340085037bac0261ea/oneview_redfish_toolkit/api/chassis.py#L22)):*
<details><summary><i>Previously, llvieira (Lucas Lima Vieira) wrote…</i></summary><blockquote>

I can make it when I implement Enclosure Chassi, if you want...
</blockquote></details>

LGTM... we can make a follow up patch with it! I guess inheritance is python way of providing multiple constructors with the same number of parameters...

---

*[oneview_redfish_toolkit/api/chassis.py, line 48 at r2](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/47#-KtmKP1Mi705ehEoKjbN:-KtmKP1Mi705ehEoKjbO:b9qltp9) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/be850b4afbba78abe3abc4340085037bac0261ea/oneview_redfish_toolkit/api/chassis.py#L48)):*
> ```Python
>         self.redfish["Id"] = hardware["uuid"]
>         self.redfish["Name"] = hardware["name"]
>         self.redfish["ChassisType"] = "Blade"
> ```

In case we need to improve this hardcoded "Blade" thing we could use "category" field from the "hardware" object to map to a redfish value. OneView responses to chassis and rack also contains this field :)

---

*[oneview_redfish_toolkit/api/chassis.py, line 52 at r2](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/47#-KtmKu2EO5hukmhXUd40:-KtmKu2EO5hukmhXUd41:b-7icsu1) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/be850b4afbba78abe3abc4340085037bac0261ea/oneview_redfish_toolkit/api/chassis.py#L52)):*
> ```Python
>         self.redfish["Model"] = hardware["model"]
>         self.redfish["SerialNumber"] = hardware["serialNumber"]
>         self.redfish["IndicatorLED"] = "Unknown"
> ```

Aren't we supposed to map this to the server hardware "status" value?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/47#-:-KtmMiJgefxPmst8NEgH:b-rrdlll)*

**Comment by llvieira** [comment]
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[oneview_redfish_toolkit/api/chassis.py, line 52 at r2](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/47#-KtmKu2EO5hukmhXUd40:-KtmPVg8XO74gxwBPXTT:bbovriy) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/be850b4afbba78abe3abc4340085037bac0261ea/oneview_redfish_toolkit/api/chassis.py#L52)):*
<details><summary><i>Previously, ricardoas (Ricardo Araújo) wrote…</i></summary><blockquote>

Aren't we supposed to map this to the server hardware "status" value?
</blockquote></details>

Yes, we should map this.
I already have a method to map the Redfish' IndicatorLED to OneView's status. I created it so I could use when I was going to create the EnclosureChassi. If you want I can send it for you.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/47)*

**Comment by danielreed** [comment]
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[oneview_redfish_toolkit/api/chassis.py, line 22 at r2](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/47#-KtIzp0p10cXn930ahjM-r2-22:-Ktmv31HgMjMPYfVCLM2:b-15825i) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/be850b4afbba78abe3abc4340085037bac0261ea/oneview_redfish_toolkit/api/chassis.py#L22)):*
<details><summary><i>Previously, ricardoas (Ricardo Araújo) wrote…</i></summary><blockquote>

LGTM... we can make a follow up patch with it! I guess inheritance is python way of providing multiple constructors with the same number of parameters...
</blockquote></details>

ok sounds good

---

*[oneview_redfish_toolkit/api/chassis.py, line 48 at r2](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/47#-KtmKP1Mi705ehEoKjbN:-KtmuCc6SidZY1g9_vdh:b-39yc85) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/be850b4afbba78abe3abc4340085037bac0261ea/oneview_redfish_toolkit/api/chassis.py#L48)):*
<details><summary><i>Previously, ricardoas (Ricardo Araújo) wrote…</i></summary><blockquote>

In case we need to improve this hardcoded "Blade" thing we could use "category" field from the "hardware" object to map to a redfish value. OneView responses to chassis and rack also contains this field :)
</blockquote></details>

I don't think we need to change this.  With the changes Lucas is proposing we will know if we are creating a new Blade/Rack/Encl chassis.

---

*[oneview_redfish_toolkit/api/chassis.py, line 52 at r2](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/47#-KtmKu2EO5hukmhXUd40:-KtmuxTKbaZT5buYXWxr:b-szqmk1) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/be850b4afbba78abe3abc4340085037bac0261ea/oneview_redfish_toolkit/api/chassis.py#L52)):*
<details><summary><i>Previously, llvieira (Lucas Lima Vieira) wrote…</i></summary><blockquote>

Yes, we should map this.
I already have a method to map the Redfish' IndicatorLED to OneView's status. I created it so I could use when I was going to create the EnclosureChassi. If you want I can send it for you.
</blockquote></details>

Yes I will add Status in the next patch

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/47)*

**Comment by llvieira** [comment]
Reviewed 1 of 4 files at r3, 4 of 4 files at r4.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/47)*

**Pull request merged**